src/catalog_xliff.cpp

```diff
@@ -595,14 +596,24 @@ class XLIFF10CatalogItem : public XLIFF12CatalogItem
     }
 };
 
-
 void XLIFF1Catalog::Parse(pugi::xml_node root)
 {
     int id = 0;
+    
     for (auto file: root.children("file"))
     {
-        m_sourceLanguage = Language::TryParse(file.attribute("source-language").value());
-        m_language = Language::TryParse(file.attribute("target-language").value());
+        // TODO: Only first `file` node atthributes are used for now
+        //       what works well in case of either all next are same within
+        //       same `xliff` or if the only single `file` node is there
+        //       (what is always the only case for downloaded from Crowdin)
+        //       But should be taken into account and improved for probable
+        //       more wide varietty of cases in the future.
+        if(id == 0)
+        {
+            m_sourceLanguage = Language::TryParse(file.attribute("source-language").value());
+            SetLanguage(Language::TryParse(file.attribute("target-language").value()));
```

Comment:
I don't think any of these leftover SetLanguage calls are actually needed anymore?

Reply:
Why if I see e.g. below in files downloaded from Crowdin?
```
<file id="25" original="/Test word file.docx" source-language="en" target-language="uk" datatype="plaintext">
```

Reply:
> Why if I see e.g. below in XLiff files downloaded from Crowdin?

I don't understand what you mean by that. I was referring to the - unneeded as far as I can tell - change from the original `m_language = Language::TryParse(file.attribute("target-language").value());` to this.

Reply:
Ah, now I got what you meant, sorry for being unapprehensive. This was part of some issue I experienced when implemented uploading files directly as XLIFF with new v2 API but unfortunately and my bad I have not commented which exactly issues it was. But do you see/suspect/feel it might create any side effect? Since I see [`XLIFF2Catalog::Parse()` uses it as well](https://github.com/tetatetit/poedit/blob/278a08250a71def13b71847c9e962309d829e90e/src/catalog_xliff.cpp#L751)

Reply:
> Since I see XLIFF2Catalog::Parse() uses it as well

Because you changed it there too… The comment applies to all such uses.

I had the grep the sources on `master` w.r.t. SetLanguage() and it modifies the XML tree. I.e. it is for external use (e.g. when the *user* changes the language) of the class, while these uses are internal during parsing.

Reply:
Sorry about confusion with that function in ` XLIFF2Catalog::Parse() ` as well. Really have forgotten about this due of lot of changes (now I see my mistake that really needed to make smaller linked PRs from the beginning of my work which would be done not against master bug against e.g. branch called `v2.4` or `crowdin-v2-api` as base. Anyway do you see any possible side effect from using `SetLanguage()` internally from inside `Parse()`. Since I definitely could not change `SetLanguage()` in 2 places just for fun :) so there definitely should be some reason to this, at least as it seemed for me. So I just afraid that to do not break any Xlifff<=>Crowdin functionality by reverting it back. That is why I am paying attention to this issue and asking you so thoroughly.

Reply:
Still addressed https://github.com/vslavik/poedit/pull/633#discussion_r442268381